src/main/kotlin/racingcar/service/RacingCarGameMachine.kt

```diff
@@ -6,11 +6,17 @@ import racingcar.service.strategy.NumberGenerateStrategy
 
 class RacingCarGameMachine(private val numberGenerate: NumberGenerateStrategy) {
 
-    fun play(carCount: Int, tryCount: Int): List<GameResultDto> {
-        var cars = List(carCount) { Car() }
+    fun play(carNames: List<String>, tryCount: Int): List<GameResultDto> {
+        var cars = carNames.map { name -> Car(name) }
         return (1..tryCount).map { stage ->
             cars = cars.map { car -> car.move(numberGenerate.generate()) }
-            cars.let { cars -> GameResultDto(stage, cars) }
+            GameResultDto(stage, cars)
         }
     }
+
+    fun getWinners(gameResults: List<GameResultDto>): List<String> {
+        val lastStage = gameResults.sortedBy { it.stage }.last()
```

Comment:
바로 다음 줄에서 사용하신 `maxByOrNull` 확장함수를 활용하도록 변경할 수 있을 것 같아요. 😃
```suggestion
        val lastStage = gameResults.maxByOrNull { it.stage } ?: throw IllegalArgumentException("Error MSG")
```